conans/client/cmd/download.py

```diff
@@ -48,11 +49,15 @@ def _download_binaries(conanfile, ref, package_ids, cache, remote_manager, remot
 
     def _download(package_id):
         pref = PackageReference(ref, package_id)
-        # TODO: cache2.0 here when specifying a package_id with conan download
-        #  we are not checking the remote for the latest package revision
-        #  maybe we want to add an argument to conan download like --update
-        check_pref_cache = cache.get_package_revisions(pref)
-        skip_download = (len(check_pref_cache) > 0)
+        try:
+            if not pref.revision:
```

Comment:
It is impossible that pref has revision here, you just constructed it above, empty revision?